src/game/Combat/ThreatManager.cpp

```diff
@@ -281,6 +281,20 @@ void ThreatContainer::modifyThreatPercent(Unit* victim, int32 threatPercent)
     }
 }
 
+void ThreatContainer::modifyAllThreatPercent(int32 threatPercent)
+{
+    ThreatList const& tList = getThreatList();
+    UnitList uList;
+
+    // store attacker
+    for (auto itr : tList)
+        uList.push_back(itr->getTarget());
+
+    // process the threat modification
+    for (auto itr : uList)
+        modifyThreatPercent(itr, threatPercent);
```

Comment:
Allow me to suggest even less iterating by not building a unit list first. You're now basically iterating over the ThreatList once to build a unit list of equal size, then you iterate over that unit list. Unless I've missed something, this seems even simpler:

```
for (auto itr: tList)
    modifyThreatPercent(itr->getTarget(), threatPercent);
```
If duplicates would be possible and should be avoided, your solution is better as long as you use a set instead of a list.

Reply:
@evil-at-wow it sound tempting but no, the modifyThreatPercent method can modify the tList storage resulting to a crash.
Yeah not so evident at first look. Its an old code that probably need some upgrade to simplify it and make it more clear.
He choose the safest way to do it, maybe there is a more optimized way to do it tho.